perspective_reddit_bot/check_mod_actions.py

```diff
@@ -51,66 +51,163 @@
-  if status_fields is not None:
-    record.update(status_fields)
-
-  log_subreddit_comments.append_records(output_path, [record])
-
-
-def fetch_reddit_comment_status(reddit, comment_id, has_mod_creds):
+# TODO: This implementation of batching is a bit complex: read_lines yields
+# sentinel values at regular intervals so that read_batches is able to yield
+# batches within max_batch_delay. The issue is that when reading from the end of
+# the file, there could be an indeterminate amount of time until the next line,
+# which would cause accumulated lines to be "stranded".
+#
+# This can almost certainly be done in a simpler way...
+
+def read_lines(input_path, stop_at_eof, yield_every_period, sentinel):
+  """Read lines from file.
+
+  Args:
+    input_path: (str) where to read from
+    stop_at_eof: (bool) returns reaching end of file if true.
+    yield_every_period: (timedelta) yields sentinel value every time period if
+        stream doesn't have contents.
+    sentinel: value yielded every yield_every_period
+
+  Yields: lines from input_path, or sentinel values
+  """
+  with open(input_path) as f:
+    last_time = datetime.utcnow()
+    while True:
+      now = datetime.utcnow()
+      where = f.tell()
+      line = f.readline()
+      if line:
+        print('.', end='')
+        sys.stdout.flush()
+        yield line
+        last_time = now
+      elif stop_at_eof:
+        print('\nread_lines stopping due to stop_at_eof.')
+        return
+      elif now - last_time > yield_every_period:
+        yield sentinel
+        last_time = now
+      else:
+        print('\n(eof, waiting {})'.format(yield_every_period))
+        time.sleep(yield_every_period.total_seconds())
+        f.seek(where)
+
+
+_PRAW_BATCH_SIZE = 100
+
+
+def read_batches(input_path, stop_at_eof, max_batch_delay):
+  """Yields batches from input_path."""
+  current_batch = []
+  last_yield_time = datetime.utcnow()
+  sentinel = object()
+  for line in read_lines(input_path, stop_at_eof, max_batch_delay, sentinel):
+    now = datetime.utcnow()
+    if line is not sentinel:
+      current_batch.append(line)
+    if (current_batch
+        and (len(current_batch) == _PRAW_BATCH_SIZE
+             or now - last_yield_time > max_batch_delay)):
+      print('batch(', len(current_batch), ')')
+      yield current_batch
+      current_batch = []
+      last_yield_time = now
+
+  # read_lines exited
+  if current_batch:
+    yield current_batch
+
+
+def read_records_with_wait(input_path, stop_at_eof, max_batch_delay,
+                           timestamp_key, wait_delta):
+  """Yields batches, waiting until all items are old enough."""
+  for batch in read_batches(input_path, stop_at_eof, max_batch_delay):
+    records = [json.loads(line) for line in batch]
+    youngest = max(r[timestamp_key] for r in records)
+    create_time_utc = datetime.strptime(youngest, '%Y%m%d_%H%M%S')
+    wait_until(create_time_utc + wait_delta)
+    yield records
+
+
+def prefix_comment_id(i):
```

Comment:
Why do we want comment_ids to begin with 't1_'?

Reply:
because `reddit.info()` requires the 'fullname' ID, but our other tools don't include the fullname when logging. added docstring  and comment.